From 640c0f866e791a226e86e31e7e42e62452b5e166 Mon Sep 17 00:00:00 2001
From: Qiankun Miao <qiankun.miao@intel.com>
Date: Wed, 5 Aug 2015 16:56:23 +0800
Subject: [PATCH] Use a fallback draw instead of CopyTexSubImage for
 copySurface

This is a workaroud for CrossWalk. Skia upstream already fixed this
issue. When CrossWalk rolls that fix, this workaroud should be removed.
---
 src/gpu/GrDrawTarget.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/gpu/GrDrawTarget.cpp b/src/gpu/GrDrawTarget.cpp
index 549ab37..5274ab6 100644
--- a/src/gpu/GrDrawTarget.cpp
+++ b/src/gpu/GrDrawTarget.cpp
@@ -440,11 +440,6 @@ bool GrDrawTarget::copySurface(GrSurface* dst,
         return true;
     }
 
-    if (this->getGpu()->canCopySurface(dst, src, clippedSrcRect, clippedDstPoint)) {
-        this->onCopySurface(dst, src, clippedSrcRect, clippedDstPoint);
-        return true;
-    }
-
     GrRenderTarget* rt = dst->asRenderTarget();
     GrTexture* tex = src->asTexture();
 
-- 
2.4.0

